src/main/java/org/antlr/intellij/plugin/preview/PreviewPanel.java

```diff
@@ -315,6 +316,15 @@ private void switchToGrammar(VirtualFile grammarFile) {
 		ANTLRv4PluginController controller = ANTLRv4PluginController.getInstance(project);
 		PreviewState previewState = controller.getPreviewState(grammarFile);
 
+		// From 1.18, automatically set the start rule name to the first rule in the grammar
+		// if none has been specified
+		if ( previewState.g!=null && previewState.startRuleName==null ) {
+			OrderedHashMap<String, Rule> rules = previewState.g.rules;
+			if (rules != null && rules.size() > 0) {
+				previewState.startRuleName = rules.getElement(0).name;
```

Comment:
Is it complicated to choose the first rule without references (now such rules marked as `unused parse rule`)?

Reply:
yeah, I couldn't find an easy way and as I mentioned in the bug, I think picking the first rule is the best bet.